### DISCUSSION TITLE
[Feature] addon multi-cluster differentiated configuration

### DISCUSSION THREAD

**Discussion by zhiyu0729** [kubevela/kubevela#5259] [Q&A]
**Is your feature request related to a problem? Please describe.**

I want to install an ingress addon.
Now I have two clusters, and I have created an LB for each cluster, I need to bind the each ingress to the corresponding LB.
Any ideas？

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by wonderflow** [comment]
Addon actually is a KubeVela Application, you just need to modify the [application template](https://github.com/kubevela/catalog/blob/master/addons/ingress-nginx/template.cue) with [multi-cluster override policy](https://kubevela.net/docs/case-studies/multi-cluster#override-default-configurations-in-clusters).

**Reply by zhiyu0729** [reply]
Well, rather than modifying the addon, I prefer to pass in parameters.

**Reply by wonderflow** [reply]
Well, than, it's an opinionated thing, I'm not sure if most of our users need this. You can send a PR to the catalog if you've any concrete ideas, I hope we don't add too many complexity.

**Reply by zhiyu0729** [reply]
It should not be to a catalog, I think this should be implemented in the core, each cluster can have different parameters

**Reply by wonderflow** [reply]
The addon already using CUE to render the application, the parameter can work defined in CUE. Can you please describe the case that don't satisfy your need? It can help us understand the feature request.

**Reply by zhiyu0729** [reply]
Suppose I have a parameter load balancing ID, cluster a is aaa, cluster b is bbb, and cluster c will be added later.
The current way is how to do it, write both the id parameters of cluster a and the id parameters of cluster b in the addon?

**Reply by wonderflow** [reply]
The case you mentioned just indicates that this is complex, you can't easily decalre that in command, maybe you have to write json such as:

```
vela addon enable ingress --cluster-config-override {"name":"clustera", "lb":"aaa"}  --cluster-config-override {"name":"clusterb", "lb":"bbb"} 
```

You can see in the example, this is not a good UX if you insist using `parameters`. In this case, why don't write an application for the addon?
